python/analyzers/HNLTreeProducer.py

```diff
@@ -19,33 +19,69 @@ def declareVariables(self, setup):
         # event variables
         self.bookEvent(self.tree)
         self.var      (self.tree, 'n_cands')
+        self.var      (self.tree, 'rho')
+        self.var      (self.tree, 'nLeptons')
+        self.var      (self.tree, 'nElectrons')
+        self.var      (self.tree, 'nMuons')
         
         # reco variables
         self.bookHNL (self.tree, 'hnl')
-        self.var     (self.tree, 'hnl_iso_abs')
-        self.var     (self.tree, 'hnl_iso_rel')
+        self.var     (self.tree, 'hnl_iso03_abs_rhoArea')
+        self.var     (self.tree, 'hnl_iso04_abs_rhoArea')
+        self.var     (self.tree, 'hnl_iso05_abs_rhoArea')
+        self.var     (self.tree, 'hnl_iso03_abs_deltaBeta')
+        self.var     (self.tree, 'hnl_iso04_abs_deltaBeta')
+        self.var     (self.tree, 'hnl_iso05_abs_deltaBeta')
+#        self.var     (self.tree, 'hnl_iso_abs_met')
+        self.var     (self.tree, 'hnl_iso03_rel_rhoArea')
+        self.var     (self.tree, 'hnl_iso04_rel_rhoArea')
+        self.var     (self.tree, 'hnl_iso05_rel_rhoArea')
+        self.var     (self.tree, 'hnl_iso03_rel_deltaBeta')
+        self.var     (self.tree, 'hnl_iso04_rel_deltaBeta')
+        self.var     (self.tree, 'hnl_iso05_rel_deltaBeta')
+#        self.var     (self.tree, 'hnl_iso_rel_met')
         
         if   self.cfg_ana.promptLepType == 'ele':
             self.bookEle (self.tree, 'l0')
+            self.var(self.tree, 'hlt_Ele27_WPTight_Gsf'        )
+            self.var(self.tree, 'hlt_Ele32_WPTight_Gsf'        )
+            self.var(self.tree, 'hlt_Ele35_WPTight_Gsf'        )
+            self.var(self.tree, 'hlt_Ele115_CaloIdVT_GsfTrkIdT')
+            self.var(self.tree, 'hlt_Ele135_CaloIdVT_GsfTrkIdT')
+
         elif self.cfg_ana.promptLepType == 'mu':
             self.bookMuon(self.tree, 'l0')
+            self.var(self.tree, 'hlt_IsoMu24'                   )
+            self.var(self.tree, 'hlt_IsoMu27'                   )
+            self.var(self.tree, 'hlt_Mu50'                      )
         else:
              print 'ERROR: prompt lepton type non specified or missing! Exit'
              exit(0)
-    
-        self.bookMuon(self.tree, 'l1' )
-        self.bookMuon(self.tree, 'l2' )
+
+        if self.cfg_ana.L1L2LeptonType == 'mm':
+            self.bookMuon(self.tree, 'l1' )
+            self.bookMuon(self.tree, 'l2' )
+        if self.cfg_ana.L1L2LeptonType == 'ee':
+            self.bookEle(self.tree, 'l1'  )
+            self.bookEle(self.tree, 'l2'  )
+        if self.cfg_ana.L1L2LeptonType == 'em':
+            self.bookEle(self.tree, 'l1'  )
+            self.bookMuon(self.tree, 'l2' )
```

Comment:
here you decide that, for cross flavours, e comes first, muon comes second.
Which should be ok, but good to know

Reply:
yes, that's why we only have eem/mem and never eme/mme